app/bot.go

```diff
@@ -193,6 +193,93 @@ func (a *App) disableUserBots(userId string) *model.AppError {
 	return nil
 }
 
+func (a *App) notifySysadminsBotDisabled(userId string) *model.AppError {
+
+	botsDisabled := *a.Config().ServiceSettings.DisableBotsWhenOwnerIsDeactivated
+
+	perPage := 20
+	options := &model.BotGetOptions{
+		OwnerId:        userId,
+		IncludeDeleted: false,
+		OnlyOrphaned:   false,
+		Page:           0,
+		PerPage:        perPage,
+	}
+	userBots, err := a.GetBots(options)
+	if err != nil {
+		return err
+	}
+
+	// user does not own bots
+	if len(userBots) == 0 {
+		return nil
+	}
+
+	// user being disabled
+	user, err := a.GetUser(userId)
+	if err != nil {
+		return err
+	}
+
+	// get sysadmins
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  perPage,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+	sysAdmins, err := a.GetUsers(userOptions)
+	if err != nil {
+		return err
+	}
+
+	// for each sysadmin, notify user that owns bots was disabled
+	for _, sysAdmin := range sysAdmins {
+		channel, appErr := a.GetOrCreateDirectChannel(sysAdmin.Id, sysAdmin.Id)
+		if appErr != nil {
+			return appErr
+		}
+
+		post := &model.Post{
+			UserId:    sysAdmin.Id,
+			ChannelId: channel.Id,
+			Message:   GetDisableBotSysadminMessage(user, userBots, botsDisabled),
+			Type:      model.POST_SYSTEM_GENERIC,
+		}
+
+		_, appErr = a.CreatePost(post, channel, false)
+		if appErr != nil {
+			return appErr
+		}
+	}
+	return nil
+}
+
+func GetDisableBotSysadminMessage(user *model.User, userBots model.BotList, botsDisabled bool) string {
+
+	infoMessage := fmt.Sprintf("For more information, see our [documentation](https://docs.mattermost.com/developer/bot-accounts.html#what-happens-when-a-user-who-owns-bot-accounts-is-disabled).")
+
+	var message, botMessage string
+	for _, bot := range userBots {
+		botMessage += fmt.Sprintf("* %v\n", bot.Username)
+	}
+
+	if botsDisabled {
+		message += fmt.Sprintf("%v was deactivated. The user managed the following bot accounts which have now been disabled.\n\n", user.Username)
+		message += botMessage
+		message += fmt.Sprintf("Do not worry - you can take ownership of each bot by enabling it at **Integrations > Bot Accounts** and creating new tokens for the bot.\n\n")
+		message += infoMessage
+	} else {
+		message += fmt.Sprintf("%v was deactivated. The user managed the following bot accounts which are still enabled.\n\n", user.Username)
+		message += botMessage
+		message += fmt.Sprintf("We strongly recommend you to take ownership of the bot by re-enabling it at **Integrations > Bot Accounts** and creating new tokens for the bot.\n\n")
+		message += fmt.Sprintf("%+v\n\n", infoMessage)
+		message += fmt.Sprintf("If you want bot accounts to disable automatically after user deactivation, set “Disable bot accounts after user deactivation” in **System Console > Custom Integrations > Bot Accounts** to true.")
```

Comment:
```suggestion
		message += fmt.Sprintf("If you want bot accounts to disable automatically after user deactivation, set “Disable bot accounts after user deactivation” in **System Console > Integrations > Bot Accounts** to true.")
```

Reply:
No problem @jasonblais!  Thanks for all the feedback.  Now waiting for the devs to give this PR a thorough scrubbing :)